src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader.cs

```diff
@@ -0,0 +1,1274 @@
+            return true;
+        }
+
+        private bool ConsumeString()
+        {
+            Debug.Assert(_buffer.Length >= _consumed + 1);
+            Debug.Assert(_buffer[_consumed] == JsonConstants.Quote);
+
+            //Create local copy to avoid bounds checks.
+            ReadOnlySpan<byte> localCopy = _buffer;
+
+            int idx = localCopy.Slice(_consumed + 1).IndexOf(JsonConstants.Quote);
+            if (idx < 0)
+            {
+                if (IsLastSpan)
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.EndOfStringNotFound);
+                else
+                    return false;
+            }
+
+            if (localCopy[idx + _consumed] != JsonConstants.ReverseSolidus)
+            {
+                localCopy = localCopy.Slice(_consumed + 1, idx);
+
+                if (localCopy.IndexOfAnyControlOrEscape() != -1)
+                {
+                    _lineBytePosition++;
+                    ValidateEscapingAndHex(localCopy);
+                    goto Done;
+                }
+
+                _lineBytePosition += idx + 1;
+
+            Done:
+                _lineBytePosition++;
+                ValueSpan = localCopy;
+                _tokenType = JsonTokenType.String;
+                _consumed += idx + 2;
+                return true;
+            }
+            else
+            {
+                return ConsumeStringWithNestedQuotes();
+            }
+        }
+
+        private bool ConsumeStringWithNestedQuotes()
+        {
+            //TODO: Optimize looking for nested quotes
+            //TODO: Avoid redoing first IndexOf search
+            int i = _consumed + 1;
+
+            Debug.Assert(_buffer.Length >= i);
+
+            while (true)
+            {
+                int counter = 0;
+                int foundIdx = _buffer.Slice(i).IndexOf(JsonConstants.Quote);
+                if (foundIdx == -1)
+                {
+                    if (IsLastSpan)
+                        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.EndOfStringNotFound);
+                    else
+                        return false;
+                }
+                if (foundIdx == 0)
+                    break;
+                for (int j = i + foundIdx - 1; j >= i; j--)
+                {
+                    if (_buffer[j] != JsonConstants.ReverseSolidus)
+                    {
+                        if ((counter & 1) == 0)
+                        {
+                            i += foundIdx;
+                            goto FoundEndOfString;
+                        }
+                        break;
+                    }
+                    else
+                        counter++;
+                }
+                i += foundIdx + 1;
+            }
+
+        FoundEndOfString:
+            int startIndex = _consumed + 1;
+            ReadOnlySpan<byte> localCopy = _buffer.Slice(startIndex, i - startIndex);
```

Comment:
Consider naming this something else, like `foundString`.  The `localCopy` in other methods currently means `saving the field to a local`, but this one is end-sliced.